tree/tree/src/TTree.cxx

```diff
@@ -1249,9 +1249,10 @@ bool CheckReshuffling(TTree &mainTree, TTree &friendTree)
 
    if ((isMainReshuffled || isFriendReshuffled) && !friendHasValidIndex) {
       const auto reshuffledTreeName = isMainReshuffled ? mainTree.GetName() : friendTree.GetName();
-      const auto msg = "Tree '%s' has the kEntriesReshuffled bit set, and cannot be used as friend nor can be added as "
-                       "a friend unless the main tree has a TTreeIndex on the friend tree '%s'. You can also unset the "
-                       "bit manually if you know what you are doing.";
+      const auto msg =
+         "Tree '%s' has the kEntriesReshuffled bit set and cannot have friends nor can be added as a friend unless the "
+         "main tree has a TTreeIndex on the friend tree '%s'. You can also unset the bit manually if you know what you "
```

Comment:
I am not understanding the wording "the main tree has a TTreeIndex on the friend tree".   If I remember correctly in order for a friend to be properly read when "out of order", the index needs to be build in the friend. as the index contains the entry number corresponding to a given `Major/Minor' (eg Run Number /  Entry Number).   So we call `BuildIndex` on the friend (referring to column with the same names in both) and the main tree does not (directly) know about the index .